### PR TITLE
Address Intellij inspection findings

### PR DESCRIPTION
**Pull request by snazy**
I let IntelliJ run all inspections on the Iceberg source tree, and it returned quite some amount of issues. A bunch are reported as issues, that require attention.

This PR addresses a lot of these inspection issues, none of the individual changes seem controversial, so I opted to push one PR with a couple commits.

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/PartitionSpec.java [hunk 1]
```diff
@@ -410,7 +410,7 @@ private void checkAndAddPartitionName(String name, Integer sourceColumnId) {
         }
       }
       Preconditions.checkArgument(
-          name != null && !name.isEmpty(), "Cannot use empty or null partition name: %s", name);
+          !name.isEmpty(), "Cannot use empty or null partition name: %s", name);
```
Comment (ajantha-bhat) [inline api/src/main/java/org/apache/iceberg/PartitionSpec.java:413]: we can remove "or null" text from message also.

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java [hunk 2]
```diff
@@ -220,16 +220,15 @@ private void writeDataWithFailOnPartition(
     final int numPartitions = 10;
     final int partitionToFail = new Random().nextInt(numPartitions);
     MapPartitionsFunction<Row, Row> failOnFirstPartitionFunc =
-        (MapPartitionsFunction<Row, Row>)
```
Comment (ajantha-bhat) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java:223]: We can fix it for other version of spark folders also.

Comment (ajantha-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java:223]: Similar comment for all Spark and Flink module related changes. Only default modules are handled currently. Maybe follow up PR also is fine.

File: core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java [hunk 3]
```diff
@@ -76,19 +76,15 @@ public static ByteBuffer longToOrderedBytes(long val, ByteBuffer reuse) {
    * ByteBuffer)}
    */
   public static ByteBuffer shortToOrderedBytes(short val, ByteBuffer reuse) {
-    ByteBuffer bytes = ByteBuffers.reuse(reuse, PRIMITIVE_BUFFER_SIZE);
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java:79]: a) should we replace contents of `intToOrderedBytes` to call `longToOrderedBytes` ?
b) Replace all other `intToOrderedBytes` to call `longToOrderedBytes` to avoid double casting (type -> int -> long)

**Review by ajantha-bhat — COMMENTED** [review]
Thanks for this awesome clean up work. It was easy to review commit by commit. 

Few minor comments. It is good to go. 
The non-default modules (of Flink and Spark) can be handled as a follow up PR.

**Comment by snazy** [comment]
I'd really prefer to keep non-IJ-autofixes out of this PR.

**Comment by nastra** [comment]
The PR is already quite large and given that we need to fix those things across all Flink/Spark versions I'd suggest to extract the Flink/Spark related things into a separate PR

File: api/src/main/java/org/apache/iceberg/PartitionSpec.java (hunk 1 above)
Comment (nastra) [reply api/src/main/java/org/apache/iceberg/PartitionSpec.java:413]: +1 on updating the error msg here

File: core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java (hunk 3 above)
Comment (ajantha-bhat) [reply core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java:79]: let me rephrase this comment. 

a) Why didn't we handle the same for `intToOrderedBytes`? It is a duplicate of `longToOrderedBytes`
b) Why are we not calling `longToOrderedBytes` in the new places? calling `intToOrderedBytes` will leads to double casting.

**Comment by ajantha-bhat** [comment]
>  The PR is already quite large and given that we need to fix those things across all Flink/Spark versions I'd suggest to extract the Flink/Spark related things into a separate PR

I think the current changes are easy to review if we review commit by commit. The backport to Flink/Spark can be handled in the follow up.

File: api/src/main/java/org/apache/iceberg/PartitionSpec.java (hunk 1 above)
Comment (nastra) [reply api/src/main/java/org/apache/iceberg/PartitionSpec.java:413]: this still needs to be addressed

**Comment by snazy** [comment]
Repeating my note from above: IMHO no _functional_ or potentially _user facing_ changes should go into this PR.
Going to rebase this once more due to the conflict.

**Comment by nastra** [comment]
should be good to go once https://github.com/apache/iceberg/pull/10583/files#r1663651011 has been addressed

**Comment by snazy** [comment]
> should be good to go once https://github.com/apache/iceberg/pull/10583/files#r1663651011 has been addressed

Sorry, it's really out of scope of this PR ("Adress IntelliJ Inspection findings").

**Comment by nastra** [comment]
> > should be good to go once https://github.com/apache/iceberg/pull/10583/files#r1663651011 has been addressed
> 
> Sorry, it's really out of scope of this PR ("Adress IntelliJ Inspection findings").

We're improving the code and the null check is being removed since it wasn't null in the first place, so what prevents us from updating `"Cannot use empty or null partition name:` to `"Cannot use empty partition name:` to also improve the error msg in order to reflect the current state? I don't see see a reason to improve this error msg in a separate PR, hence this can be handled while touching the same code line

**Comment by snazy** [comment]
> We're improving the code and the null check is being removed

"improving code" is very broad - and the null-check was never hit. Really - this PR is meant to address the intellij findings and nothing more.

File: api/src/main/java/org/apache/iceberg/PartitionSpec.java (hunk 1 above)
Comment (nastra) [inline api/src/main/java/org/apache/iceberg/PartitionSpec.java:413]:
```suggestion
          !name.isEmpty(), "Cannot use empty partition name: %s", name);
```

**Review by nastra — APPROVED** [review]
LGTM, @snazy can you please follow up and backport those changes to older Flink/Spark versions?
